test/node/src/migration.test.ts

```diff
@@ -792,6 +844,48 @@ for (const dialect of DIALECTS) {
           'migration1',
         ])
       })
+
+      describe('Migrate up should work when timestamps are equal', () => {
+        // The following lines of code simulate a situation where the migrations would have the
+        // timestamp
+
+        let originalToIsoString: typeof Date.prototype.toISOString
+
+        before(() => {
+          originalToIsoString = Date.prototype.toISOString
+          const defaultDateIsoString = new Date(2024, 0, 11).toISOString()
+          Date.prototype.toISOString = () => defaultDateIsoString
+        })
+
+        after(() => {
+          // Reset to originalToIsoString function so that upcoming tests are not affected
+          Date.prototype.toISOString = originalToIsoString
+        })
```

Comment:
This is not clean, too coupled with implementation detail.
Use [sinon fake timers](https://sinonjs.org/releases/latest/fake-timers/) instead.